src/main/java/dk/bec/polonez/reservationsystem/service/OfferService.java

```diff
@@ -35,36 +41,79 @@ public OfferService(OfferRepository offerRepository, FeatureRepository featureRe
     }
 
     public OfferDto getById(long id) {
-        Offer offer = offerRepository.getById(id);
-        return modelMapper.map(offer, OfferDto.class);
+        Optional<Offer> offer = offerRepository.findById(id);
+        return modelMapper.map(offer.orElseThrow(
+                () -> new ResponseStatusException(HttpStatus.NOT_FOUND, "Offer with id=" + id + " not found")), OfferDto.class);
     }
 
     public OfferDto addOffer(CreateOfferDto offerDto) {
-        Offer offer = modelMapper.map(offerDto, Offer.class);
-        Offer savedOffer = offerRepository.save(offer);
-        return modelMapper.map(savedOffer, OfferDto.class);
+        if (authService.isPoLoggedIn() || authService.isAdminLoggedIn()) {
+            Offer offer = modelMapper.map(offerDto, Offer.class);
+            offer.setOwner(authService.getCurrentUser());
+            Offer savedOffer = offerRepository.save(offer);
+            return modelMapper.map(savedOffer, OfferDto.class);
+        } else {
+            throw new AuthenticationException(HttpStatus.FORBIDDEN, "Access denied, log as Place Owner or System Admin");
```

Comment:
**advice** Business exception shoule be thrown here. ```OfferService``` should abstract away from REST and HTTP-specific details.

Reply:
My comment was similar to our comments in previous pull-requests, that's why I didn't explain fully, sorry. Generally you should throw your own custom exception in your service (like ```NoAccessToOfferCreationException```) and then you should implement custom exception mapper (in a separate class) that would map your custom exception to specific HTTP error code. By doing this we are separating pure business logic from HTTP/REST details.